commands/bash_completion.go

```diff
@@ -23,18 +24,20 @@ var bashcompletionCmd = &cobra.Command{
 This currently only works on Bash version 4, and is hidden
 pending a merge of https://github.com/spf13/cobra/pull/520.`,
 	Hidden: true,
-	Run:    runBashcompletion,
+	RunE:   runBashcompletion,
 }
 
-func runBashcompletion(cmd *cobra.Command, args []string) {
+func runBashcompletion(cmd *cobra.Command, args []string) error {
 	if len(args) < 1 {
 		fmt.Println("Please provide filename for bash completion")
-		return
+		return errors.New("Please provide filename for bash completion")
```

Comment:
You can use `fmt.Errorf("error text..")` instead of pulling in the `error` import. Again a general comment that applies across the PR.